Assets/Scripts/Game/Entities/PlayerEntity.cs

```diff
@@ -575,7 +575,7 @@ public void AssignItems(SaveTree saveTree)
                         newItem.TrappedSoulType = MobileTypes.None;
                 }
                 // Add to local inventory or wagon
-                if (containerRecord.IsWagon)
+                if (containerRecord.RecordRoot.SpriteIndex == 4) // Wagon containers have sprite index 4
```

Comment:
The property IsWagon also performs the (recordRoot.SpriteIndex == 4) check. Is there a reason you removed that property?

I would rather this being performed in the ContainerRecordData and exposed as IsWagon, rather than having the PlayerEntity check for sprite index directly.

Reply:
No big reason. First I took out the wagon specific checks in favor of identifying all container types, and then when I saw there was that one spot using the wagon check I figured since it was just one spot it would be OK to just have it directly do the check itself, as I'm not sure that we'll ever need to do the check elsewhere. But I'll go ahead and stick it in again.